### ISSUE TITLE
Support wsgi in twistd on python3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8276]
|<img alt="uspike's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| uspike reported|
|-|-|
|Trac ID|trac#8276|
|Type|defect|
|Created|2016-04-07 01:05:10Z|

With the completion of [#7993](https://github.com/twisted/twisted/issues/7993), the python2-only import of `twisted.web.wsgi `in twisted/web/tap.py was not updated, causing import errors when running `twistd web --wsgi` on python3.

Specific line here: https://github.com/twisted/twisted/blob/616a7422eee0053f69002a878e2f317bc8aaa63c/twisted/web/tap.py#L23

```
(venv3) jordan@enigma:~/src/twisted$ twistd -n web --wsgi my.app
Traceback (most recent call last):
  File "/home/jordan/src/twisted/venv3/bin/twistd", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/jordan/src/twisted/bin/twistd", line 18, in <module>
    run()
  File "/home/jordan/src/twisted/twisted/scripts/twistd.py", line 29, in run
    app.run(runApp, ServerOptions)
  File "/home/jordan/src/twisted/twisted/application/app.py", line 638, in run
    config.parseOptions()
  File "/home/jordan/src/twisted/twisted/application/app.py", line 605, in parseOptions
    usage.Options.parseOptions(self, options)
  File "/home/jordan/src/twisted/twisted/python/usage.py", line 267, in parseOptions
    self.subOptions.parseOptions(rest)
  File "/home/jordan/src/twisted/twisted/python/usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "/home/jordan/src/twisted/twisted/python/usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "/home/jordan/src/twisted/twisted/web/tap.py", line 155, in opt_wsgi
    self['root'] = wsgi.WSGIResource(reactor, pool, application)
NameError: name 'wsgi' is not defined

```
adiroiban:
   For me it make sense to just think that [#7993](https://github.com/twisted/twisted/issues/7993) did the wsgi heavy lifting and consider the job done and create a new ticket to take care of twistd integration. 
   Please create a new ticket + patch + test which checkes that wsgi is available in twisted in py3 

It seems like tests are already written, but not enabled in python3.  Attached is a patch to import/enable wsgi.

Attachments:

* [wsgi.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/406/406be5d7bdf47ebe0bd179182b3cf580c4dd2923/825fb5e946f24cba8a88efe5fe4a13bf128d79f8.patch) (1999 bytes) - added by uspike on 2016-04-07 01:06:22Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__8276 8276
type__defect defect
reporter__uspike uspike
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1459991110253154 1459991110253154
changetime__1469405409398183 1469405409398183
version__None None
owner__uspike uspike
cc__adiroiban
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

uspike. Is your patch ready for review?


Please let us know that it is ready for review by adding the 'review' keyword to this ticket.

Since this patch introduces a changes which is visible to end users, please create a NEWS file fragment as described here https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles

In this way, the next release will include a note to inform everybody that twisted wsgi is not available on py3

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

See https://github.com/twisted/twisted/pull/404

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub set status to `closed`|
|-|-|

In changeset 1e5d96a7c9ee8ce7fd4e798b4a877f169ad0e686
```
#!CommitTicketReference repository="" revision="1e5d96a7c9ee8ce7fd4e798b4a877f169ad0e686"
Merge pull request #404 from alex/patch-1

Author: alex
Reviewer: hawkowl
Fixes: #8276
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

Awesome!

I feel like maybe this should have had a topfile though?  Users might care that `--wsgi` works now?
